### PR TITLE
Token deletion confirmation

### PR DESCRIPTION
**Pull request by kof**
Closes https://github.com/webstudio-is/webstudio-builder/issues/1672

## Description

We need to show a confirmation dialog when user hits delete to explain what will happen

## Steps for reproduction

1. add a token
2. click delete

Screenshot 2023-05-31 at 15 25 43

## Code Review

- [ ] hi @TrySound , I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-builder/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `builder/env-check.js` if mandatory

### CODE REVIEW

**Comment by taylornowotny** [comment]
Looks great!

I would prefer to remove the word "entire" and "Yes, " as they are unnecessary. Not critical though.

File: apps/builder/app/builder/features/style-panel/style-source-section.tsx [hunk 1]
```diff
@@ -341,53 +352,118 @@ export const StyleSourcesSection = () => {
...
+            selectedStyleSourceSelectorStore.set({
+              styleSourceId: id,
+            });
+          }
+        }}
+        onChangeItem={(item) => {
+          renameStyleSource(item.id, item.label);
+        }}
+      />
+      {tokenToDelete && (
```
Comment (TrySound) [inline apps/builder/app/builder/features/style-panel/style-source-section.tsx:411]: Why not use open prop?

File: apps/builder/app/builder/features/style-panel/style-source-section.tsx [hunk 2]
```diff
@@ -341,53 +352,118 @@ export const StyleSourcesSection = () => {
...
+          convertLocalStyleSourceToToken(id);
+          setEditingItemId(id);
+        }}
+        onRemoveItem={(id) => {
+          removeStyleSourceFromInstance(id);
+        }}
+        onDeleteItem={(id) => {
+          const token = availableStyleSources.find(
+            (source) => source.id === id
+          );
```
Comment (TrySound) [inline apps/builder/app/builder/features/style-panel/style-source-section.tsx:384]:
```
const styleSources = styleSourcesStore.get())
const token = styleSources.get(id)
```

File: apps/builder/app/builder/features/style-panel/style-source-section.tsx (hunk 1 above)
Comment (kof) [reply apps/builder/app/builder/features/style-panel/style-source-section.tsx:411]: dunno, felt semantically like I only want to render the component when we are in the right state, what's the downside?

Comment (TrySound) [reply apps/builder/app/builder/features/style-panel/style-source-section.tsx:411]: Inconsistent usage and lack of transition when there is any.

File: apps/builder/app/builder/features/style-panel/style-source-section.tsx (hunk 2 above)
Comment (kof) [reply apps/builder/app/builder/features/style-panel/style-source-section.tsx:384]: done

File: apps/builder/app/builder/features/style-panel/style-source-section.tsx (hunk 1 above)
Comment (kof) [reply apps/builder/app/builder/features/style-panel/style-source-section.tsx:411]: done

**Pull request merged**